File: app/assets/javascripts/models/wine.js
WC.Models.Wine = Backbone.RelationalModel.extend({
  url: '/wines/',

  schema: {
    name: 'Text',
    country: 'Text',
    color: {type: 'Select', options: ['red', 'white', 'rose']},
    vintage: 'Number',
    quantity: 'Number'
  },

  relations: [{
    type: 'HasMany',
    key: 'tasting_notes',
    relatedModel: 'WC.Models.TastingNote',
    collectionType: 'WC.Collections.TastingNotes',
    includeInJSON: false
  }]
});

File: app/models/wine.rb
class Wine < ActiveRecord::Base
  attr_accessible :name, :country, :color, :vintage, :quantity, :user_id

  has_many :tasting_notes
  has_many :storage_locations
  has_many :bins, through: :storage_locations
  belongs_to :user

  validates :vintage, allow_nil: true
end